mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/ErrorRecoveryIndexationTask.java

```diff
@@ -19,15 +19,111 @@
 
 package org.apache.mailbox.tools.indexer;
 
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
+import javax.inject.Inject;
+
+import org.apache.james.mailbox.MessageUid;
 import org.apache.james.mailbox.indexer.ReIndexingExecutionFailures;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.server.task.json.dto.TaskDTO;
 import org.apache.james.task.Task;
 import org.apache.james.task.TaskExecutionDetails;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.github.steveash.guavate.Guavate;
+
 public class ErrorRecoveryIndexationTask implements Task {
     private static final String PREVIOUS_FAILURES_INDEXING = "ErrorRecoveryIndexation";
 
+    public static class ErrorRecoveryIndexationTaskDTO implements TaskDTO {
+
+        public static ErrorRecoveryIndexationTaskDTO of(String type, ErrorRecoveryIndexationTask task) {
```

Comment:
I think the type is always be `PREVIOUS_FAILURES_INDEXING` because this DTO is for serving `ErrorRecoveryIndexationTask`, so can you remove that argument?

Reply:
Yes you are right i should have declared the dto module as constant into the task.